docs/content/en/docs/Development/ide.md

```diff
@@ -0,0 +1,316 @@
+---
+title: IDE configuration
+weight: 50
+---
+
```

Comment:
Maybe just start with #VSCode? I get that `heir-lsp` and `clangd` are technically more generic (And maybe add a statement to that extent to the "we only support vscode disclaimer"?) but I think it'd be easier for most people if the page worked well top-down as a way to set-up vscode with all the necessary extensions, depenencies and settings.json snippets, with non-vscode users simply reading the same instructions and skipping stuff not relevant for their setup.

Reply:
Forcing everyone to read through a bunch of VSCode config to find the stuff relevant for their IDE seems... a bit egocentric :) I can't make sense of the VSCode config, and I don't expect you to make sense of the neovim config.

Also, I don't think the HEIR docs is the right place to teach people about their IDE (e.g., how to run a debugger, or how to use the bazel extension), and it would be better to just link to the dedicated documentation for those tools, or write a blog post that makes it clear when it was published so that people don't expect it to stay up to date as various external tools change beyond our control.

Reply:
Fair enough! I actually don't even use the bazel integration in vscode (it's waaay too slow to update the target list) and run bazel from the terminal, too, but it's still nice to have the formatting/lsp/etc :) 

Maybe we should (in the next iteration, let's get this merged first) just collect all the settings.json stuff into a single "dump this into your settings.json" thing. In addition to the clangd stuff, there's overriding the python formatter with pyink and (not yet documented but highly recommended)  excluding /bazel and /external from python linting/analysis and the files watcher, as well as setting the lsp path.